packages/amplify_core/lib/src/types/storage/copy_options.dart

```diff
@@ -1,10 +1,38 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import 'package:amplify_core/amplify_core.dart';
+
 /// {@template amplify_core.storage.copy_options}
-/// Configurable options to initiate a [StorageCopyRequest].
+/// Configurable options for `Amplify.Storage.copy`.
 /// {@endtemplate}
-abstract class StorageCopyOptions {
+class StorageCopyOptions
+    with AWSEquatable<StorageCopyOptions>, AWSSerializable, AWSDebuggable {
   /// {@macro amplify_core.storage.copy_options}
-  const StorageCopyOptions();
+  const StorageCopyOptions({this.pluginOptions});
+
+  /// plugin specific options for `Amplify.Storage.copy`.
+  final StorageCopyPluginOptions? pluginOptions;
+
+  @override
+  List<Object?> get props => [pluginOptions];
+
+  @override
+  String get runtimeTypeName => 'StorageCopyOptions';
+
+  @override
+  Object? toJson() {
+    return {'pluginOptions': pluginOptions?.toJson()};
+  }
+}
+
+/// {@template amplify_core.storage.copy_plugin_options}
+/// plugin specific options for `Amplify.Storage.copy`.
+/// {@endtemplate}
+abstract class StorageCopyPluginOptions
```

Comment:
Are we going to create the abstract type for plugin options of all operations?
I only saw `StorageCopyPluginOptions` throughout this PR

Reply:
yes, I'm doing it for all operations. added for copy operation first to confirm the impl details before doing it for all.